README.md

```diff
@@ -71,7 +71,65 @@ To update the expected output of the tests, use the `--update-expected-output` f
+>This is relevant for internal development to publish development images of kontrol with modified kontrol chagnes and retains fixed upstream dependencies.
+The usecase for this workflow is intended to make testing changes of kontrol needed for use in testing CI or in other downstream workflows without needing to publish changes or PRs first. 
+
+Intent is to reduce friction of needing custom builds and avoiding lengthy upstream changes and PRs.
+
+
+
+### Build Kontrol with Kup and Specific Dependency Override
+Relevant to this workflow [kup-build-kontrol.yml](.github/workflows/kontrol-push-unfixed-deps.yml)
+> This is relevant for internal development to publish development images of kontrol for use in kaas or a dockerized test environment.
+Use the workflow [Kup Build Kontrol](.github/workflows/kup-build-kontrol.yml) to publish a custom version of Kontrol for use in CI and [KaaS](https://kaas.runtimeverification.com/).
+[See KUP docs for more information](https://github.com/runtimeverification/kup/blob/master/src/kup/install-help.md#kup-install----override) 
+
+#### Using Kup 
+Relevant dependency options are shown below and can be listed using `kup list kontrol --inputs`  
+For example: 
+```
+Inputs:
+├── k-framework - follows kevm/k-framework
+├── kevm - github:runtimeverification/evm-semantics (6c2526b)
+│   ├── blockchain-k-plugin - github:runtimeverification/blockchain-k-plugin (c9264b2)
+│   │   ├── k-framework - github:runtimeverification/k (5d1ccd5)
+│   │   │   ├── haskell-backend - github:runtimeverification/haskell-backend (d933d5c)
+│   │   │   │   └── rv-utils - follows kevm/blockchain-k-plugin/k-framework/llvm-backend/rv-utils
+│   │   │   ├── llvm-backend - github:runtimeverification/llvm-backend (37b1dd9)
+│   │   │   │   ├── immer-src - github:runtimeverification/immer (4b0914f)
+│   │   │   │   └── rv-utils - github:runtimeverification/rv-nix-tools (a650588)
+│   │   │   └── rv-utils - follows kevm/blockchain-k-plugin/k-framework/llvm-backend/rv-utils
+│   │   └── rv-utils - follows kevm/blockchain-k-plugin/k-framework/rv-utils
+│   ├── haskell-backend - follows kevm/k-framework/haskell-backend
+│   ├── k-framework - github:runtimeverification/k (81bcc24)
+│   │   ├── haskell-backend - github:runtimeverification/haskell-backend (786c780)
+│   │   │   └── rv-utils - follows kevm/k-framework/llvm-backend/rv-utils
+│   │   ├── llvm-backend - github:runtimeverification/llvm-backend (d5eab4b)
+│   │   │   ├── immer-src - github:runtimeverification/immer (4b0914f)
+│   │   │   └── rv-utils - github:runtimeverification/rv-nix-tools (a650588)
+│   │   └── rv-utils - follows kevm/k-framework/llvm-backend/rv-utils
+│   └── rv-utils - follows kevm/k-framework/rv-utils
+└── rv-utils - follows kevm/rv-utils
+```
+Now run a build using kup and specific dependency overrides:    
+
+`kup install kontrol --override kevm/haskell-backend "hash/branch name" --override kevm/k-framework/haskell-backend "hash/branch name"`
```

Comment:
What do you think? 
branch name -> branch_name
```suggestion
`kup install kontrol --override kevm/haskell-backend "hash/branch_name" --override kevm/k-framework/haskell-backend "hash/branch_name"`
```

Reply:
Working on it [over here](https://github.com/runtimeverification/devops-actions/actions/workflows/kup-build-kontrol.yml) so I can test it